File: fcca_go/landunit.go
package main

type LandUnit struct {
	Unit
}

func newLandUnit(x, y int, military *Military) *LandUnit {
	return &LandUnit {
		Unit {
			x: x,
			y: y,
			military: military,
		},
	}
}

func (unit *LandUnit) Move(int dir) {
}

func (unit *LandUnit) x() { return unit.x }
func (unit *LandUnit) y() { return unit.y }

File: cociv/cociv/landunit.rb
class LandUnit < Unit

protected

  def move_ok?(fx, fy, log=false)
    tile = @game[fx,fy]
    return false if not generic_move_ok? fx, fy, log
    if [Ocean].include? tile.terrain
      $log.debug 'Movement rejected because land units can\'t move over water.' if log
      return false 
    end

    return true
  end

end
